applications/asset_tracker_v2/src/cloud/cloud_codec/lwm2m/lwm2m_codec_helpers.h

```diff
@@ -0,0 +1,157 @@
+ *  @retval 0 If successful, otherwise a negative value indicating the reason of failure.
+ */
+int lwm2m_codec_helpers_get_configuration_object(struct cloud_data_cfg *cfg);
+
+/** @brief Set GNSS data.
+ *
+ *  @param[in] gnss Pointer to structure that contains GNSS data.
+ *
+ *  @retval 0 If successful, otherwise a negative value indicating the reason of failure.
+ *  @return -ENODATA if the queued flag present in the input structure is false.
+ */
+int lwm2m_codec_helpers_set_gnss_data(struct cloud_data_gnss *gnss);
+
+/** @brief Set environmental sensor data.
+ *
+ *  @param[in] sensor Pointer to structure that contains environmental sensor data.
+ *
+ *  @retval 0 If successful, otherwise a negative value indicating the reason of failure.
+ *  @return -ENODATA if the queued flag present in the input structure is false.
+ */
+int lwm2m_codec_helpers_set_sensor_data(struct cloud_data_sensors *sensor);
+
+/** @brief Set modem dynamic data.
+ *
+ *  @param[in] modem_dynamic Pointer to structure that contains dynamic modem data.
+ *
+ *  @retval 0 If successful, otherwise a negative value indicating the reason of failure.
+ *  @return -ENODATA if the queued flag present in the input structure is false.
+ */
+int lwm2m_codec_helpers_set_modem_dynamic_data(struct cloud_data_modem_dynamic *modem_dynamic);
+
+/** @brief Set modem static data.
+ *
+ *  @param[in] modem_static Pointer to structure that contains static modem data.
+ *
+ *  @retval 0 If successful, otherwise a negative value indicating the reason of failure.
+ *  @return -ENODATA if the queued flag present in the input structure is false.
+ */
+int lwm2m_codec_helpers_set_modem_static_data(struct cloud_data_modem_static *modem_static);
+
+/** @brief Set battery data.
+ *
+ *  @param[in] battery Pointer to structure that contains battery data.
+ *
+ *  @retval 0 If successful, otherwise a negative value indicating the reason of failure.
+ *  @return -ENODATA if the queued flag present in the input structure is false.
+ */
+int lwm2m_codec_helpers_set_battery_data(struct cloud_data_battery *battery);
+
+/** @brief Set User Interface (UI) data.
+ *
+ *  @param[in] user_interface Pointer to structure that contains user_interface data.
+ *
+ *  @retval 0 If successful, otherwise a negative value indicating the reason of failure.
+ *  @return -ENODATA if the queued flag present in the input structure is false.
+ */
+int lwm2m_codec_helpers_set_user_interface_data(struct cloud_data_ui *user_interface);
+
+/** @brief Set neighbor cell data.
+ *
+ *  @param[in] neighbor_cells Pointer to structure that contains neighbor cell data.
+ *
+ *  @retval 0 If successful, otherwise a negative value indicating the reason of failure.
+ *  @return -ENODATA if the queued flag present in the input structure is false.
+ */
+int lwm2m_codec_helpers_set_neighbor_cell_data(struct cloud_data_neighbor_cells *neighbor_cells);
+
+/** @brief Convenience API used to generate path lists with reference to objects.
+ *
```

Comment:
My lwm2m knowledge fails me when I try to understand this. When does a application typically need to call this function?